### ISSUE TITLE
Support for new Palemoon 31.0 version

### ISSUE THREAD

**Issue by albertdev** [JustOff/github-wc-polyfill#66]
Hi,

After the dust has settled a bit on the Palemoon dev team shenanigans, the promised 31 version got released. Can the supported version be bumped in the manifest? I noticed that some parts of Github still refuse to load on v31.

----

Also, with regards to the confusion in #59 and #60 about the Palemoon GUID: it seems that Moonchild has reverted to the situation as it was in 29.1.x or older:

https://forum.palemoon.org/viewtopic.php?f=46&t=28172

**Comment by SeaHOH** [comment]
Wait ... until it is really get ready.

**Comment by AroKol78** [comment]
> Wait ... until it is really get ready.

is it about that `??` ? (i suppose)
"Implement nullish coalescing operator (??)" -> `https://repo.palemoon.org/MoonchildProductions/UXP/issues/1894`

new UXP and forks (ST52):
Implement `queueMicrotask()`
Implement optional chaining `?.`

**Comment by albertdev** [comment]
I'm sorry, I'm not sure I understand the replies here.

Problem is that using Palemoon 31.0 to surf to Github will put a bunch of errors in the browser console, e.g. "TypeError: window.customElements is undefined".

I can't install this addin because Palemoon complains:
![image](https://user-images.githubusercontent.com/1451362/168448234-82ab3952-c658-4b61-9e83-2382efedecf3.png)

> Wait ... until it is really get ready.

Do you mean that you're waiting for the next Palemoon update because it's not yet "ready"? E.g. do they still need to add something you need or fix some open bugs?

Or do you mean that you're fully aware of the Palemoon 31 being released and that I should just have waited instead of creating ticket? I promise I will not create a duplicate or ping you all the time, I mainly wanted to check to see if this is on the radar.

Thank you for helping to maintain this plugin in any case.

----
EDIT: It's not super critical for me, I simply wait with upgrading Palemoon on the computers where I need Github the most.

**Comment by AroKol78** [comment]
if you need it immediately, please use the advice included #62 (https://github.com/JustOff/github-wc-polyfill/issues/62#issuecomment-1112063626 and https://github.com/JustOff/github-wc-polyfill/issues/62#issuecomment-1112138632)

i was thinking of using new functions instead of emulated example `queueMicrotask()`
https://github.com/JustOff/github-wc-polyfill/blob/6dfe469c02866f2b74d995172613eb6f3ecb6242/bootstrap.js#L166
using `??` and `?.` could also be helpful = "github-wc-polyfill 2.0" (of course I can be wrong)

**Comment by SeaHOH** [comment]
1. It's now PM 31.0.0 is incompatible with many addons, and will be fixed in next release.
    see https://repo.palemoon.org/moonchildproductions/uxp/issues/953
1. I need time to upgrade my browser/profile, then I can continue work for this matter.

**Comment by Vangelis66** [comment]
> I was thinking of using **new functions instead of emulated** example `queueMicrotask()`

... This will be a real head-scratcher to properly implement...

- Pale Moon `31.x.x` and **latest** New Moon `28.10.6a1` have it **implemented natively** - only **PM** is supported officially by this extension.
- **Latest** Serpent `52.9.0` has this also **implemented natively**, but it's not (the browser) officially supported by the extension; OTOH, latest official **Basilisk** `52.9.2022.01.27` (now sort of an **abandonware**...) **is** officially supported by the extension, but it (the browser) doesn't support natively `queueMicrotask()` ... 
- Since I'm not following closely/at all **SeaMonkey**'s development, am unsure about present (or planned future) support of `queueMicrotask()` in that browser...

My real question is hence: 
Would having `queueMicrotask()` **polyfilled inside this extension** (the case **now**) cause any detrimental effect when the extension is installed and used on a browser with **native** `queueMicrotask()` support?

**Comment by rofl0r** [comment]
> Would having queueMicrotask() polyfilled inside this extension (the case now) cause any detrimental effect when the extension is installed and used on a browser with native queueMicrotask() support?

i recall having seen polyfills that check whether a function/object is defined and only add it if not. it should be trivial to do the same.

**Comment by linuxrocks123** [comment]
Just FYI the change I made in the rejected PR #64 works absolutely fine for me so anyone who wants to use it please look it up.

**Comment by Gordon-Dry** [comment]
> Just FYI the change I made in the rejected PR #64 works absolutely fine for me so anyone who wants to use it please look it up.

This is the simplest approach as no code change is needed at all and of course I did it manually for myself as soon as PM 31 was installed.
I even don't need Github for it.

**Comment by AroKol78** [comment]
>... This will be a real head-scratcher to properly implement...

> **Latest** Serpent `52.9.0` has this also **implemented natively**, but it's not (the browser) officially supported by the extension; OTOH, latest official **Basilisk** `52.9.2022.01.27` (now sort of an **abandonware**...) **is** officially supported by the extension, but it (the browser) doesn't support natively `queueMicrotask()` ...


Basilisk EoL - officially no longer supported `https://forum.palemoon.org/viewtopic.php?f=25&t=28309`
so only the Basilisk forks and PM,SM (Serpent52,NM28XP,IceApe,Iceweasel and other miracles) - **all new UXP**
so I don't see the problem (this is what it looks like from my side)

**Comment by Ben-Fenner** [comment]
Just here to say I'm patiently awaiting this extension to support PM 31.

**Comment by ghost** [comment]
Me too 😊however, it looks like we have to wait for the next browser update in June, if I got it right.

**Comment by SeaHOH** [comment]
Hi, guys~
For some private reasons, I will get rid of the development of this addon. Maybe [martok/palefill](https://github.com/martok/palefill) is a available alternative for you.

**Comment by AroKol78** [comment]
**SeaHOH**

Thank you for your time spent on guiding us through this period of big changes, we wouldn't have been able to do it without you. Thank you once again.

(Dziękuję za Twój poświęcony czas na przeprowadzenie nas w okresie dużych zmian, bez Ciebie nie daliśmy by sobie rady.
Raz jeszcze dziękuje.)

**Comment by Ben-Fenner** [comment]
Yes, thank you for your efforts and letting us know the status of this extension.
Safe travels!

**Comment by sebma** [comment]
@SeaHOH 
> For some private reasons, I will get rid of the development of this addon.

Hi, does this mean this project is dying ?

**Comment by Sa-Ja-Di** [comment]
Why is nobody forking it here on Github and continuing developing it? This add-on don't need to die if other people still would be interested into it.

@SeaHOH just look for someone else who maintains it, no reason to let it end like that.

**Comment by Ben-Fenner** [comment]
@Sa-Ja-Di I'm glad you've volunteered to fork this extension and keep it going. Let us know where to find it when you're ready.

**Comment by SeaHOH** [comment]
> > For some private reasons, I will get rid of the development of this addon.
> 
> Hi, does this mean this project is dying ?

@sebma I don't know, but the owner is in trouble now.

> just look for someone else who maintains it, no reason to let it end like that.

@Sa-Ja-Di I have no full-control permission of this repo.

If you like this addon, fork it is a good solution to go on.

**Comment by Sa-Ja-Di** [comment]
@Ben-Fenner I am not forking anything as I don't have any skills, I just mentioned someone still could

@SeaHOH well if someone with skills would, it would be nice, but I am not qualified in doing any coding or maintenance tasks, I am just a user and report issues

**Comment by hipocentrum** [comment]
You can learn, I believe in you

14 cze 2022, 11:15 od ***@***.***:

**Comment by faithanalog** [comment]
the recommendation of palefill is working good for me for this and also gitlab. thanks for all the work you've done on this, and here's to a bright future with palefill. To all here, go try out https://github.com/martok/palefill as SeaHOH recommended, it has support for github and others and looks to be a good option for adding support to more websites in the future.

**Comment by Vangelis66** [comment]
Pale Moon `v31.1.0` was released on **June 7th 2022**; this is **now**, i.e. as I type this, the **minimum** PM version one should be on... 
At the end of June, **GitHub unleashed newer scripts**, which contain **both** the Optional Chaining (?.) and Nullish Coalescing (??) operators; the first was already implemented in PM `v31.0.0`, but the latter only in `v31.1.0`.

**PM** `v31.1.0` + **gh-wc-polyfill** `v1.2.19` **force-installed** should make GitHub (mostly) **usable** 😜 again.
There's still some "annoyance", tracked in #68, that requires you to reload (internal) Github pages 😞 ; a [workaround](https://github.com/JustOff/github-wc-polyfill/issues/68#issuecomment-1167531061) and a [third-party offered XPI file](https://github.com/JustOff/github-wc-polyfill/issues/68#issuecomment-1171379830) both aim to mitigate that annoyance... 😄
